lib/reporters/writers/console.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/**
+ * Save timer references (mochajs/mocha#237).
+ */
+var setTimeout = global.setTimeout;
+
+/**
+ * Provides methods used for output by console-oriented reporters.
+ *
+ * @description
+ * Not meant to be used directly.
+ *
+ * @package
+ * @mixin ConsoleWriter
+ */
+var ConsoleWriter = function() {
+  var buffer = '';
+
+  /**
+   * Writes string to reporter output.
+   *
+   * @package
+   * @param {string} str - String to write to the reporter output.
+   * @returns {this}
+   * @chainable
+   */
+  this.write = function(str) {
+    if (str) {
+      buffer += str; // Store until `console.log` invoked
+    }
+    return this;
+  };
+
+  /**
+   * Writes newline-terminated string to reporter output.
+   *
+   * @package
+   * @param {string} str - String to write to the reporter output.
+   * @returns {this}
+   * @chainable
+   */
+  this.writeln = function(str) {
+    this.write(str);
+    console.log(buffer);
```

Comment:
Worth saving a reference like the timer? (let users stub)

Reply:
I hadn't bothered integrating with your PR at this point, but it would be done in this one file for all reporters implementing `ReportWriter`. It's quite honestly trivial to implement for a reporter; it's writing the unit tests that takes all the time...